pkg/cloud/service.go

```diff
@@ -41,32 +47,122 @@ type Service struct {
+		return nil, err
+	}
+	ga, err := compute.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		return nil, err
+	}
+
+	nsGA, err := networkservicesga.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		return nil, err
+	}
+	nsBeta, err := networkservicesbeta.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		return nil, err
+	}
+
+	svc := &Service{
+		GA:                  ga,
+		Alpha:               alpha,
+		Beta:                beta,
+		NetworkServicesGA:   nsGA.Projects.Locations,
+		NetworkServicesBeta: nsBeta.Projects.Locations,
+		ProjectRouter:       pr,
+		RateLimiter:         rl,
+	}
+
+	return svc, nil
+}
+
 // wrapOperation wraps a GCE anyOP in a version generic operation type.
-func (s *Service) wrapOperation(anyOp interface{}) (operation, error) {
+func (s *Service) wrapOperation(anyOp any) (operation, error) {
 	switch o := anyOp.(type) {
 	case *ga.Operation:
 		r, err := ParseResourceURL(o.SelfLink)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("wrapOperation: %w", err)
 		}
-		return &gaOperation{s: s, projectID: r.ProjectID, key: r.Key}, nil
+		return &gaOperation{
+			s:         s,
+			projectID: r.ProjectID,
+			key:       r.Key,
+		}, nil
 	case *alpha.Operation:
 		r, err := ParseResourceURL(o.SelfLink)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("wrapOperation: %w", err)
 		}
-		return &alphaOperation{s: s, projectID: r.ProjectID, key: r.Key}, nil
+		return &alphaOperation{
+			s:         s,
+			projectID: r.ProjectID,
+			key:       r.Key,
+		}, nil
 	case *beta.Operation:
 		r, err := ParseResourceURL(o.SelfLink)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("wrapOperation: %w", err)
+		}
+		return &betaOperation{
+			s: s, projectID: r.ProjectID,
+			key: r.Key,
+		}, nil
+	case *networkservices.Operation:
+		result, err := parseNetworkServiceOpURL(o.Name)
+		if err != nil {
+			return nil, fmt.Errorf("wrapOperation: %w", err)
+		}
+		return &networkServicesOperation{
+			s:         s,
+			projectID: result.projectID,
+			key:       result.key,
+		}, nil
+	case *networkservicesbeta.Operation:
+		result, err := parseNetworkServiceOpURL(o.Name)
+		if err != nil {
+			return nil, fmt.Errorf("wrapOperation: %w", err)
 		}
-		return &betaOperation{s: s, projectID: r.ProjectID, key: r.Key}, nil
+		// Reuse the GA operation stream for Beta.
+		return &networkServicesOperation{
+			s:         s,
+			projectID: result.projectID,
+			key:       result.key,
+		}, nil
 	default:
 		return nil, fmt.Errorf("invalid type %T", anyOp)
 	}
 }
 
+type networkServiceOpURLParseResult struct {
+	projectID string
+	key       *meta.Key
+}
+
+// parseNetworkServiceOpURL parses the URL of the network services operation.
+// This is different than the `compute` API paths.
+func parseNetworkServiceOpURL(name string) (*networkServiceOpURLParseResult, error) {
+	// Format: projects/<projectID>/locations/global/operations/<Name>
```

Comment:
do we need to do a sanity check that split[0] == projects, split[2] == locations , .... ?

Reply:
adding.